apps/backend/src/users/users.service.ts

```diff
@@ -21,4 +25,55 @@ export class UsersService {
   remove(id: number) {
     return `This action removes a #${id} user`;
   }
+
+  async addFavoriteDrink(userId: string, favoriteDrinkId: string): Promise<User> {
+    const user = await this.prisma.user.findUnique({
+      where: { authSchId: userId },
+      include: { favouriteDrinks: true },
+    });
+
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    if (user.favouriteDrinks.length >= 3) {
+      throw new BadRequestException('A user can have only 3 favorite drinks.');
+    }
+
+    if (user.favouriteDrinks.some((drink) => drink.id === favoriteDrinkId)) {
+      throw new BadRequestException('The drink is already in the user list of favorite drinks.');
+    }
+
+    const updatedUser = await this.prisma.user.update({
+      where: { authSchId: userId },
+      data: { favouriteDrinks: { connect: { id: favoriteDrinkId } } },
+      include: { favouriteDrinks: true },
+    });
```

Comment:
If the user provides a valid uuid, but no drinks exist with that ID, this throws an exception that isn't handled

Reply:
I came up with 2 solutions. I used the cleaner and more readable one, but this is more optimized:
 try {
      const updatedUser = await this.prisma.user.update({
        where: { authSchId: userId },
        data: { favouriteDrinks: { connect: { id: favoriteDrinkId } } },
        include: { favouriteDrinks: true },
      });
      return updatedUser;
    } catch (error) {
      if (error.code === 'P2025') {
        throw new NotFoundException('Drink not found');
      }
      throw error;
    }